x-pack/plugins/security_solution/cypress/integration/detection_rules/prebuilt_rules.spec.ts

```diff
@@ -59,7 +59,16 @@ describe('Prebuilt rules', () => {
 
       changeRowsPerPageTo(rowsPerPage);
 
-      cy.get(SHOWING_RULES_TEXT).should('have.text', `Showing ${expectedNumberOfRules} rules`);
+      cy.request({ url: '/api/detection_engine/rules/_find' }).then(({ body }) => {
+        // Assert the total number of loaded rules equals the expected number of in-memory rules
+        expect(body.total).to.equal(rawRules.length);
```

Comment:
If you don't specify request parameters for this call, pagination options will default to 1st page and 20 rules per page:

https://github.com/elastic/kibana/blob/0212338e4e3ec9db5cfb4e93ad1850a156f0ea9b/x-pack/plugins/security_solution/common/detection_engine/schemas/request/find_rules_schema.ts#L14-L29

I think here we need to fetch _all_ rules by specifying `per_page: 10000`. I wonder how this test passed on CI. Maybe I'm missing something?

Reply:
Sorry, please disregard that 🤦 `total` is the total number of rules matching the filter, and not the number of rules requested for a given page.